src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,159 @@
+ */
+public class Storage {
+
+    /** Directory file. */
+    private final File directory;
+
+    /** Text file. */
+    private final File textFile;
+
+    /** UI for storage to display messages. */
+    private final Ui ui;
+
+    /** Task list for storage to access. */
+    private final TaskList taskList;
+
+    /**
+     * Creates a storage for Duke's tasks data.
+     *
+     * @param directoryPath Path of the directory.
+     * @param textFilePath Path of the text file.
+     * @param ui UI.
+     * @param taskList List of tasks.
+     */
+    public Storage(String directoryPath, String textFilePath, Ui ui, TaskList taskList) {
+        this.directory = new File(directoryPath);
+        this.textFile = new File(textFilePath);
+        this.ui = ui;
+        this.taskList = taskList;
+    }
+
+    /**
+     * Gets the UI.
+     *
+     * @return UI.
+     */
+    public Ui getUi() {
+        return ui;
+    }
+
+    /**
+     * Gets the list of tasks.
+     *
+     * @return List of tasks.
+     */
+    public TaskList getTaskList() {
+        return taskList;
+    }
+
+    /**
+     * Checks if duke.txt file exists in the data directory of the root folder.
+     * If it exists, load the data into Duke.
+     * If it does not exist, create one.
+     */
+    public void checkSaveFile() {
+        if (directory.exists()) {
+            if (textFile.exists()) {
+                readTextFile();
+            } else {
+                createTextFile();
+            }
+        } else {
+            createDataDirectory();
+            createTextFile();
+        }
+    }
+
+    /**
+     * Creates a data directory in the root folder.
+     */
+    private void createDataDirectory() {
+        boolean hasCreatedDir = directory.mkdir();
+        if (hasCreatedDir) {
+            this.ui.printMessage("A data directory has been created in the root folder.");
+        } else {
+            this.ui.printWarning("Failed to create a data directory.");
+        }
+    }
+
+    /**
+     * Creates duke.txt in the data directory of the root folder.
+     */
+    private void createTextFile() {
+        String failMessage = "Failed to create duke.txt.";
+        try {
+            boolean hasCreatedFile = textFile.createNewFile();
+            if (hasCreatedFile) {
+                this.ui.printMessage("duke.txt has been created in the data directory of the root folder.");
+            } else {
+                this.ui.printWarning(failMessage);
+            }
+        } catch (IOException e) {
+            this.ui.printWarning(failMessage);
+        }
+    }
+
+    /**
+     * Reads duke.txt and adds tasks into the task list.
+     */
+    private void readTextFile() {
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(textFile));
+            String line = reader.readLine();
+            while (line != null) {
+                String[] arr = line.split(" / ");
+                Task task;
```

Comment:
Consider initializing the task variable first.

Reply:
Done. Thanks.